src/waves/waves.js

```diff
@@ -34,7 +34,9 @@ export class MdWaves {
     }
 
     this.attributeManager.addClasses(classes);
-    Waves.attach(this.element);
+    if (!(typeof window.require === 'function' && typeof window.require.specified === 'function' && window.require.specified('waves'))) {
```

Comment:
I am not sure why it works. Its just an observation.

window.require vs window.requirejs.. maybe it has something to do how materialize amd is build?

The remove task sound interesting :smile: